### PR TITLE
Test framework

### PR DESCRIPTION
**Pull request by mdavis95**
Add test framework that allows running a yaml based test configuration.  The yaml allows defining queries and JS based test expression on the results of the queries.

Refactor tools out of the zulia-server project into zulia-tools which now includes the zuliatest tool.

bump common compress to 1.26.1 (from 1.25.0) for security fix
bump flapdoodle mongo to 4.12.2 (from 4.11.1)
bump jansi to 2.4.1 (from 2.4.0)

### CODE REVIEW

**Comment by mdavis95** [comment]
To run:

zuliatest --testConfig /path/to/sample_testing.yaml --testOutput /path/to/test_output.csv

**Review by payammeyer — COMMENTED** [review]
Code looks good and works. I have a couple questions/comments. Also, I wish the test file could have been kind of type safe, like if the indexName carried over down for each query and auto-completed. Also, do we want to be able to have `indexes` there and federate? like `titleStar` for three indexes.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -15,18 +15,18 @@ annontations = { module = "org.jetbrains:annotations", version = "24.0.0" }
 awssdk-s3 = { module = "software.amazon.awssdk:s3" }
```
[the remark this answers is not in the document]
Comment (mdavis95) [reply gradle/libs.versions.toml:15]: Do we know what version we want to match?

File: zulia-cmd-shared/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+plugins {
+    `java-library`
+}
+
+description = "Zulia CMD Shared"
+
+dependencies {
+    annotationProcessor(libs.picocli.codegen)
+    api(project(":zulia-common"))
```
[the remark this answers is not in the document]
Comment (mdavis95) [reply zulia-cmd-shared/build.gradle.kts:9]: Looks like it becomes `api(projects.zuliaCommon)`

**Comment by mdavis95** [comment]
Looks like we might be able to give a JSON schema to the YAML file for autocomplete.  I need to dig a little.

I was considering the multiple indexes but I was torn.

**Pull request merged**